provisioning_client/src/prov_device_ll_client.c

```diff
@@ -1118,29 +1118,43 @@ void Prov_Device_LL_DoWork(PROV_DEVICE_LL_HANDLE handle)
         }
         if (prov_info->is_connected || prov_info->prov_state == CLIENT_STATE_ERROR)
         {
+            tickcounter_ms_t current_time = 0;
+
             switch (prov_info->prov_state)
             {
                 case CLIENT_STATE_REGISTER_SEND:
-                    /* Codes_SRS_PROV_CLIENT_07_013: [ CLIENT_STATE_REGISTER_SEND which shall construct an initial call to the service with endorsement information ] */
-                    if (prov_info->prov_transport_protocol->prov_transport_register(prov_info->transport_handle, prov_transport_process_json_reply, prov_transport_create_json_payload, prov_info) != 0)
+                    if (tickcounter_get_current_ms(prov_info->tick_counter, &current_time) != 0)
                     {
-                        LogError("Failure registering device");
-                        if (prov_info->error_reason == PROV_DEVICE_RESULT_OK)
-                        {
-                            prov_info->error_reason = PROV_DEVICE_RESULT_TRANSPORT;
-                        }
+                        LogError("Failure getting the current time");
```

Comment:
Re Nit above: we could adjust `current_time` to be in seconds right after getting the tick ms time.

Reply:
Keeping it as milliseconds and changing retry_after_Value to milliseconds creates better solution.